src/databricks/labs/dqx/llm/pk_identifier.py

```diff
@@ -0,0 +1,706 @@
+        """Build the final table definition string."""
+        table_definition = "{\n" + ",\n".join(definition_lines) + "\n}"
+        if existing_pk:
+            table_definition += f"\n-- Existing Primary Key: {existing_pk}"
+        return table_definition
+
+    def get_table_definition(self, table: str) -> str:
+        """Retrieve table definition using Spark SQL DESCRIBE commands."""
+        if not self.spark:
+            raise ValueError("Spark session not available")
+
+        try:
+            print(f"🔍 Retrieving schema for table: {table}")
+
+            definition_lines = self._get_table_columns(table)
+            existing_pk = self._get_existing_primary_key(table)
+            table_definition = self._build_table_definition_string(definition_lines, existing_pk)
+
+            print("✅ Table definition retrieved successfully")
+            return table_definition
+
+        except (ValueError, RuntimeError) as e:
+            logger.error(f"Error retrieving table definition for {table}: {e}")
+            raise
+        except (AttributeError, TypeError, KeyError) as e:
+            logger.error(f"Unexpected error retrieving table definition for {table}: {e}")
+            raise RuntimeError(f"Failed to retrieve table definition: {e}") from e
+
+    def _execute_duplicate_check_query(self, table: str, pk_columns: list[str]) -> tuple[bool, int, Any]:
+        """Execute the duplicate check query and return results."""
+        pk_cols_str = ", ".join([f"`{col}`" for col in pk_columns])
+        print(f"🔍 Checking for duplicates in {table} using columns: {pk_cols_str}")
+
+        duplicate_query = f"""
+        SELECT {pk_cols_str}, COUNT(*) as duplicate_count
+        FROM {table}
+        GROUP BY {pk_cols_str}
+        HAVING COUNT(*) > 1
+        """
+
+        duplicate_result = self.spark.sql(duplicate_query)
+        duplicates_df = duplicate_result.toPandas()
+        return len(duplicates_df) > 0, len(duplicates_df), duplicates_df
+
+    @staticmethod
+    def _report_duplicate_results(
+        has_duplicates: bool, duplicate_count: int, pk_columns: list[str], duplicates_df=None
+    ):
+        """Report the results of duplicate checking."""
+        if has_duplicates and duplicates_df is not None:
+            total_duplicate_records = duplicates_df['duplicate_count'].sum()
+            logger.warning(
+                f"Found {duplicate_count} duplicate key combinations affecting {total_duplicate_records} total records"
+            )
+            print(f"⚠️  Found {duplicate_count} duplicate combinations for: {', '.join(pk_columns)}")
+            if len(duplicates_df) > 0:
+                print("Sample duplicates:")
+                print(duplicates_df.head().to_string(index=False))
+        else:
+            logger.info(f"No duplicates found for predicted primary key: {', '.join(pk_columns)}")
+            print(f"✅ No duplicates found for: {', '.join(pk_columns)}")
+
+    def check_duplicates(
```

Comment:
Move this method to `PrimaryKeyDetector`. It's not the responsibility of a class that handles table metadat to check for duplciates